UnitTests/CommonTestUtils/AsyncTestHelper.cs

```diff
@@ -18,31 +18,56 @@ public static TimeSpan UnexpectedTimeout
 
         public static void RunAndWaitForPendingOperations(Func<Task> asyncMethod, CancellationToken cancellationToken)
         {
+            if (ThreadHelper.JoinableTaskFactory == null)
+            {
+                throw new InvalidOperationException($"Run & wait is pointless without {nameof(ThreadHelper.JoinableTaskFactory)}.");
+            }
+
             ThreadHelper.JoinableTaskFactory.Run(asyncMethod);
 
             WaitForPendingOperations(cancellationToken);
         }
 
         public static T RunAndWaitForPendingOperations<T>(Func<Task<T>> asyncMethod, CancellationToken cancellationToken)
         {
+            if (ThreadHelper.JoinableTaskFactory == null)
+            {
+                throw new InvalidOperationException($"Run & wait is pointless without {nameof(ThreadHelper.JoinableTaskFactory)}.");
+            }
+
             var result = ThreadHelper.JoinableTaskFactory.Run(asyncMethod);
 
             WaitForPendingOperations(cancellationToken);
 
             return result;
         }
 
-        public static void WaitForPendingOperations(TimeSpan timeout)
+        public static void WaitForPendingOperations(TimeSpan timeout, bool throwOnTimeout = true)
```

Comment:
💡 The changes to this method are unnecessary. There are no valid places where `throwOnTimeout` can be false.